lib/page-object.rb

```diff
@@ -53,19 +54,20 @@ module PageObject
   # a method named initialize_accessors if it exists. Upon initialization of 
   # the page it will call a method named initialize_page if it exists.
   #
-  # @param [Watir::Browser or Selenium::WebDriver::Driver] the platform browser to use
+  # @param [Watir::Browser, Watir::HTMLElement or Selenium::WebDriver::Driver, Selenium::WebDriver::Element] the platform browser/element to use
   # @param [bool] open the page if page_url is set
   #
-  def initialize(browser, visit=false)
+  def initialize(root, visit=false)
     initialize_accessors if respond_to?(:initialize_accessors)
-    initialize_browser(browser)
+    initialize_browser(root)
     goto if visit && respond_to?(:goto)
     initialize_page if respond_to?(:initialize_page)
   end
 
-  def initialize_browser(browser)
-    @browser = browser
-    include_platform_driver(browser)
+  def initialize_browser(root)
```

Comment:
Set browser from the browser or element. The root is the element if an element is provided, or nil otherwise. This is done so that when asking for the root element, we can provide the main html tag at runtime instead of at initialize.